Impoppable Scripts for all Expert Maps

Janders797 opened the pull request:
A lot of changes, including
- Scan codes for every key
- Abilities and Powers (limited selection, more coming). Could be coded more elegantly...
- MouseRests default position won't move the mouse, so it doesn't screw with PlaceASAP
- A script for every Expert Map on Impoppable

also:
- A command that checks for Double-Cash mode and skips WaitForRound on most Expert Impoppable scripts (allowing you to save about 30-40 seconds per run)

Janders797 committed:
Use scan codes for everything, added power and special

Janders797 committed:
added ability buttons for Ben and Sniper

Janders797 committed:
Added buttons to access power shop + cashdrop

Janders797 committed:
restructured expert maps

Janders797 committed:
Added Open, close, ability and power

Open also tracks which tower was opened last

Janders797 committed:
Included MouseRest in WaitForUpgrade, Added WaitForAbility

Janders797 committed:
Added impoppable strats for all expert maps

while not all are perfect, and some require the use of a power, they can finish impoppable somewhat consistently, and should make you more money than buying cashdrop should lose you.

Janders797 committed:
updated Dark Castle Impoppable

Not relying on Rounds at all makes Dark Castle much easier to run with lower amounts of MK.

Janders797 committed:
Actual key.ahk

I apparently uploaded the wrong keys.ahk...

Janders797 committed:
Made MouseMove on WaitForRound conditional

The old version screwed with Place ASAP

Janders797 committed:
Fixed Dark Dungeons Winrate

Janders797 committed:
Added Double Cash image

Janders797 committed:
Added a Check for Double Cash

Double Cash lets you run some maps quicker than normally planned. This can save up to 40 seconds on impoppable runs

Janders797 committed:
Added rounds to Bloody Puddles

Janders797 committed:
added rounds to Dark Castle

Janders797 committed:
Added rounds to ouch

Janders797 committed:
Rounds + small adjustments to Dark Dungeons

Janders797 committed:
Rounds and small adjustments to Flooded Valley

Janders797 committed:
Deleted unnecessary files

Janders797 committed:
Deleted unnecessary files

Janders797 committed:
Deleted unnecessary files

Janders797 committed:
Deleted unnecessary files

Janders797 committed:
Deleted unnecessary files

Janders797 committed:
Deleted unnecessary files

Janders797 committed:
Deleted unnecessary files

Janders797 committed:
Deleted unnecessary files

Janders797 committed:
Deleted unnecessary files

Janders797 committed:
Deleted unnecessary files

Janders797 committed:
Added Rounds to Infernal

Janders797 committed:
searching smaller area for Double Cash

Searching on Full Screen lead to malfuctioning on Infernal

Janders797 committed:
Added Rounds to Muddy Puddles

Janders797 committed:
Added Rounds to Workshop

Janders797 committed:
Added Rounds to Ravine

Janders797 committed:
Added Rounds to Quad

Janders797 committed:
Fixed comments

Janders797 committed:
Updating Rounds on other Waits

Separated UpdateRound from WaitForRound, coded it to check a different location if a tower was opened on the left side, added if to other WaitFor-functions, including WaitForVictoryOrDefeat so it shows the correct Round on the defeat message if it loses after completing the rest of the script.

Janders797 committed:
WaitForRound and MouseMove adjustments

Valokoodari commented:
It's probably going to be at least a week still before I can properly read through this and do some testing. Sorry for the wait.

Janders797 commented:
> It's probably going to be at least a week still before I can properly read through this and do some testing. Sorry for the wait.

No r

Valokoodari commented:
Sorry for this really long wait but I have now realized that I don't really have the chance of properly testing this at the moment so I think it's for the best if this stays as an independent fork for the time being. I am thinking of adding a link to your fork to the readme and maybe a pinned message on Discord or something like that so people who really want these impoppable scripts can at least find then.

Also I suggest that you add a config option for using and buying of powers as monkey money can be purchased with real money and it's always possible that something goes wrong in a way which drains all of someones monkey money and/or powers.

Janders797 committed:
Geraldo support, aim, recenter, secondary targeting, etc

Also fixed an issue with round recognition

Janders797 committed:
Dark Path

Janders797 committed:
Book Wyrm Etienne

Janders797 committed:
more ability pics added

Janders797 committed:
Alternate Logs now works with MK

Can also be sped up with Double-Cash

Janders797 committed:
adjusted main

Janders797 committed:
Fix hero selection in 39.x

Janders797 committed:
Power Settings

Added option to not use Powers, so people don't accidentally drain all their Powers/Monkey Money on Impoppable maps that they can't win either way.

Janders797 committed:
Adjusted Power()

Added option to not use powers, so people don't accidentally drain all their powers/Monkey Money on Impoppable maps that they can't win either way.

Janders797 committed:
Fixed Dark Castle Chimps for patch 39.1

Janders797 committed:
Double Cash can now disable powers

Impoppable scripts generally work under Double Cash without the need of powers

Janders797 committed:
No more Cashdrop during Double Cash

Janders797 committed:
No more Cashdrop during Double Cash

Janders797 committed:
No more Cashdrop during Double Cash

Janders797 committed:
No more Cashdrop during Double Cash

Janders797 committed:
No more Cashdrop during Double Cash

Janders797 committed:
No more Cashdrop during Double Cash

Janders797 committed:
Placing Tower ASAP now also updates the Round

Janders797 committed:
Dark Castle Hard works without full Monkey Knowledge

Changing nothing about the general Path, only changed the placements and upgrades to be ASAP (and rearranged the starting placement order).
This means you don't need full MK to finish this mode, and people shouldn't run into the "Why does the Script pause"-issue anymore.

Janders797 committed:
Fixed CheckLevelUp to not interact with ASAP

CheckLevelUp left the cursor in the top left corner, meaning Place ASAP didn't work at all.

Janders797 committed:
Dark Castle Alternate works without full Monkey Knowledge

Same changes as with Dark Castle Hard.

Janders797 committed:
Slight adjustment to Dark Castle Impoppable

Janders797 committed:
Map selection

Janders797 committed:
Added Glacial TrailEasy and Impoppable

Janders797 committed:
adjusted mouserest

Janders797 committed:
mouserest + aim fix

Janders797 committed:
added mode support

Janders797 committed:
fixed targeting, added and removed some comments

Janders797 committed:
fixed targeting

Janders797 committed:
fixed druid targeting

Janders797 committed:
fixed druid targeting

Janders797 committed:
fixed druid targeting

Janders797 committed:
fixed druid targeting

Janders797 committed:
fixed druid targeting

Janders797 committed:
fixed druid targeting

Janders797 committed:
fixed druid targeting

Janders797 committed:
fixed druid targeting

Janders797 committed:
Hero change button fixed

Janders797 committed:
Hero selection fixed

Janders797 committed:
Deleted unnecessary file

Janders797 committed:
Fix round detection with snow

Janders797 committed:
Fix delay in defeat handling - Valokoodari

Janders797 committed:
Merge branch 'main' of https://github.com/Janders797/btd6-ahk

Janders797 committed:
adjusted mouse movement between waits

Janders797 committed:
Dark Castle Chimps is now all-ASAP

Janders797 committed:
Dark Castle is now all-ASAP (fixed the upgrades)

Janders797 committed:
reverted round recognition

Janders797 committed:
Merge branch 'main' of https://github.com/Janders797/btd6-ahk

Janders797 committed:
Update map selection for 41.0 (Valokoodari)

Janders797 committed:
Fix birthday event map selection for 41.0 (Valokoodari)

Janders797 committed:
Return home from collection event screen (Valokoodari)

Janders797 committed:
Fixed Glacial Trail Impoppable (1337 Gamer God)

Janders797 committed:
Fixed Glacial Trail Impoppable (part 2)

M4Marvin commented:
Most Expert Impoppable scripts dont work because Ben just got nerfed.
